### PR TITLE
Add no clobber and put bash options on their own line

### PR DESCRIPTION
**Pull request by mahesh-panchal**
- Adds the bash option no clobber
- Put each option on it's own line and explain what it does.
## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/pipeline-template/nextflow.config [hunk 1]
```diff
@@ -220,8 +220,15 @@ env {
...
 }
 
-// Capture exit codes from upstream processes when piping
-process.shell = ['/bin/bash', '-euo', 'pipefail']
+// Set bash options
+process.shell = """\
+bash
+
+set -e # Exit if a tool returns a non-zero status/exit code
+set -u # Treat unset variables and parameters as an error
```
Comment (nvnieuwk) [inline nf_core/pipeline-template/nextflow.config:228]: `-u` and `-e` are already used by Nextflow by default. I could only find one mention of it in the documentation (https://www.nextflow.io/docs/latest/process.html#process-directive-shell), but I checked in one of my runs and it's correct

Comment (nvnieuwk) [reply nf_core/pipeline-template/nextflow.config:228]: Oh I didn't know that, thanks for pointing that out!

Comment (nvnieuwk) [reply nf_core/pipeline-template/nextflow.config:228]: Oh I didn't know that, thanks for pointing that out!

**Review by nvnieuwk — COMMENTED** [review]
LGTM but I'd rather have someone else look at this too since I'm not that familiar with this

**Comment by mirpedrol** [comment]
Hi @mahesh-panchal, do you have a link to docs about the `no clobber` option?

**Comment by mahesh-panchal** [comment]
https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html

**Comment by mirpedrol** [comment]
Thanks! Could this break any existing pipelines which might be using this redirection?

**Comment by mahesh-panchal** [comment]
I'm not aware of any modules that are purposely overwriting files. We normally add checks to prevent that.

**Review by mirpedrol — APPROVED** [review]
Thanks for double checking, looks good to me then 🙂

**Pull request merged**